app/assets/javascripts/course/components/CoursePage.js

```diff
@@ -32,12 +39,102 @@ class CoursePage extends React.Component {
     })
   }
 
+  changeDisplayedSubsectionStartingFromLastComponent(id) {
+    const path = APIRoutes.getSubsectionPath(id)
+
+    request.get(path, (response) => {
+      var indexOfLastComponent = response.subsection.components.length - 1
+      this.setState({ displayed_component: response.subsection.components[indexOfLastComponent]})
+      this.setState({ displayed_subsection: response.subsection })
+    }, (error) => {
+      console.log(error)
+    })
+  }
+
+  changeDisplayedComponent(id) {
+    var targetComponent = null
+
+    for (var component of this.state.displayed_subsection.components) {
+      if (component.id == id) {
+        targetComponent = component
```

Comment:
You can probably just use this to find something in a list:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find